### PR TITLE
feat: Add support for regex macros

### PR DESCRIPTION
**Pull request by alex-barstow**
## Description
This PR does two things:
- Adds support for replacing multiple macros that match a regular expression (see full description in readme changes)
- Simplifies and fixes a design flaw in the macro name override logic, specifically for pageVariable macros

### CODE REVIEW

File: src/macros.js [hunk 1]
```diff
@@ -165,10 +158,22 @@ const getPageVariableMacros = function(string, defaults) {
...
     const resolvedMacroName = overrides.hasOwnProperty(macroName) ? overrides[macroName] : macroName;
 
-    string = string.split(resolvedMacroName).join(uriEncodeIfNeeded(macros[macroName], uriEncode));
+    if (resolvedMacroName.startsWith('r:')) {
+      try {
+        const regex = new RegExp(resolvedMacroName.slice(2), 'g');
+
+        string = string.replace(regex, uriEncodeIfNeeded(macros[macroName], uriEncode));
+      } catch (error) {
+        videojs.log.warn('Unable to replace regex macro. Please verify that you are passing valid regex.');
```
Comment (philjhale) [inline src/macros.js:170]: I'm always in favour of making error messages as explicit and useful as possible. What is the effect of being unable to replace the regex macro? What won't work?

Also, is it worth outputting the offending regex?

Comment (Essk) [reply src/macros.js:170]: I agree that outputting the input as seen by this function would be helpful to users who may well be programatically generating their regex strings.

**Review by Essk — COMMENTED** [review]
This looks great overall, happy to approve once @philjhale's comment has been answered.

File: src/macros.js (hunk 1 above)
Comment (alex-barstow) [reply src/macros.js]: Thx, it now outputs the offending regex string

**Review by Essk — APPROVED** [review]

**Pull request merged**